nl/nl_linux.go

```diff
@@ -798,7 +798,13 @@ func (s *NetlinkSocket) SetExtAck(enable bool) error {
 		enableN = 1
 	}
 
-	return unix.SetsockoptInt(int(s.fd), unix.SOL_NETLINK, unix.NETLINK_EXT_ACK, enableN)
+	var err error
+	if err = unix.SetsockoptInt(int(s.fd), unix.SOL_NETLINK, unix.NETLINK_EXT_ACK, enableN); err == unix.ENOPROTOOPT && !enable {
```

Comment:
Why suppressing the error only in the `enable == false` case?

Reply:
If the caller requires this option to present, i.e. they are setting to true, then it is important to communicate the error back. In the false case, since they don't want the option anyway, the error doesn't matter.